src/Simplify.cpp

```diff
@@ -308,12 +309,28 @@ class Simplify : public IRMutator {
         if (const int64_t *i = as_const_int(e)) {
             *min_val = *max_val = *i;
             return true;
+        } else if (t.is_uint() && is_const(e) && t.is_scalar()
```

Comment:
I think this should be implemented in `as_const_int`. It's technically correct (as long as the uint isn't greater than the max of a signed int64), but perhaps confusing to some code that uses `as_const_int`?